### PR TITLE
Add UnivariatePlotter class with various plotting methods for univarate models

### PR DESCRIPTION
**Pull request by Pjieter**
This pull request introduces a new `UnivariatePlotter` class in `src/drippy/univariate.py`, providing a comprehensive suite of plotting functions for univariate (y = constant + error) models. The class encapsulates several statistical visualization methods, making it easier to analyze and visualize sequential data. The main changes are grouped below by theme.

### New plotting functionality

* Added the `UnivariatePlotter` class, which provides methods for sequence plots, lag plots, histograms, normal probability plots, Weibull probability plots, probability plots for arbitrary distributions, Box-Cox normality plots, PPCC plots, and bootstrap plots. This centralizes and standardizes univariate statistical visualization.

### Statistical analysis enhancements

* Implemented advanced plots such as PPCC (Probability Plot Correlation Coefficient), Weibull probability, and Box-Cox normality plots, supporting deeper distributional analysis and model assessment.
* Included bootstrap plotting for statistical inference, allowing users to visualize the variability and standard error of statistics (e.g., mean) estimated from the data.

### Usability improvements

* All plotting methods accept optional `fig` and `ax` arguments for flexible integration with existing matplotlib figures and axes, and return these objects for further customization.
* Added input validation and error handling for axes shapes in

## Summary by CodeRabbit

* **New Features**
  * Added UnivariatePlotter — a univariate analysis toolkit with sequence, lag, histogram, Q‑Q/probability, Weibull, PPCC, Box‑Cox normality, bootstrap, and combined four‑plot diagnostics.
* **Documentation**
  * Added developer guidance covering plotting patterns, usage conventions, testing workflow, and contribution notes.
* **Tests**
  * Added extensive test suite validating inputs, plotting outputs, edge cases, and resource cleanup.

### CODE REVIEW

File: tests/test_univariate.py [hunk 1]
```diff
@@ -0,0 +1,1007 @@
...
+            assert len(patches) > 0
+
+    def test_with_integer_bins(self, plotter):
+        """Test histogram plot with specific integer bins."""
+        _, ax = plotter.histogram_plot(bins=10)
+
+        patches = ax.patches
+        # 10 bins specified
+        assert len(patches) == 10  # noqa: PLR2004
+
```
[the remark this answers is not in the document]
Comment (Pjieter) [reply tests/test_univariate.py:313]: @coderabbitai We want to know when the binning algorithm would change, so this would be a good indicator. We will leave it like this

File: src/drippy/univariate.py [hunk 2]
```diff
@@ -0,0 +1,594 @@
...
+            density=True,
+            label=variability_label,
+        )
+
+        ax.set_xlabel(f"Bootstrap {statistic.__name__} values")
+        ax.set_ylabel("Density")
+        ax.set_title("Bootstrap Distribution")
+        ax.legend()
+        fig.tight_layout()
+        return fig, ax
```
[the remark this answers is not in the document]
Comment (Pjieter) [reply src/drippy/univariate.py:594]: We'll fix that if it ever becomes an issue.

File: src/drippy/univariate.py [hunk 3]
```diff
@@ -0,0 +1,594 @@
...
+                The figure and axes containing the plot.
+        """
+        fig, ax = get_figure_and_axes(fig, ax)
+
+        sp.stats.probplot(self.y, dist=distribution, plot=ax)
+        ax.set_ylabel("Ordered Values")
+        ax.set_xlabel("Theoretical Quantiles")
+        ax.set_title(f"{distribution.capitalize()} Probability Plot")
+        fig.tight_layout()
+        return fig, ax
```
[the remark this answers is not in the document]
Comment (Pjieter) [reply src/drippy/univariate.py:462]: I think it is clear enough

**Comment by Pjieter** [comment]
@coderabbitai review
